plugins/ksp/src/org/jetbrains/kotlin/ksp/symbol/impl/utils.kt

```diff
@@ -246,4 +245,15 @@ internal fun FunctionDescriptor.toKSFunctionDeclaration(): KSFunctionDeclaration
         is PsiMethod -> KSFunctionDeclarationJavaImpl.getCached(psi)
         else -> throw IllegalStateException("unexpected psi: ${psi.javaClass}")
     }
+}
+
+internal fun DeclarationDescriptor.findPsi(): PsiElement? {
+    // For synthetic members.
+    if ((this is CallableMemberDescriptor) && this.kind == CallableMemberDescriptor.Kind.DECLARATION) return null
```

Comment:
Should it be SYNTHETIC?

Or simply `!=` and remove the following `if`? Finding a psi from super classes doesn't sound right to me, in our use cases.

Reply:
kind property is only available on CallableMemberDescriptor thus this check. There is no such kind as `SYNTHETIC`